File: src/main/java/io/github/jakejmattson/touchcontrol/display/ImagePanel.java
package io.github.jakejmattson.touchcontrol.display;

import javax.swing.JPanel;
import java.awt.*;
import java.awt.image.BufferedImage;

/**
 * Panel - holds image to display in GUI.
 *
 * @author JakeJMattson
 */
class ImagePanel extends JPanel
{
	private BufferedImage image;

	void setImage(BufferedImage image)
	{
		this.image = image;
	}

	@Override
	public void paintComponent(Graphics g)
	{
		super.paintComponent(g);

		if (image != null)
		{
			int width = image.getWidth();
			int height = image.getHeight();

			g.drawImage(image, 0, 0, width, height, null);
			setPreferredSize(new Dimension(width, height));
		}
	}
}

File: src/main/kotlin/io/github/jakejmattson/touchcontrol/display/ImagePanel.kt
package io.github.jakejmattson.touchcontrol.display

import java.awt.*
import java.awt.image.BufferedImage
import javax.swing.JPanel

/**
 * Panel - holds image to display in GUI.
 *
 * @author JakeJMattson
 */
internal class ImagePanel : JPanel() {
    private var image: BufferedImage? = null

    fun setImage(image: BufferedImage) {
        this.image = image
    }

    public override fun paintComponent(g: Graphics) {
        super.paintComponent(g)

        image ?: return

        val width = image!!.width
        val height = image!!.height

        g.drawImage(image, 0, 0, width, height, null)
        preferredSize = Dimension(width, height)
    }
}